src/components/notes/new-route.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect } from 'react'
+import { useClient } from 'cozy-client'
+import { createNoteDocument, generateReturnUrlToNotesIndex } from 'lib/utils'
+import Spinner from 'cozy-ui/transpiled/react/Spinner'
+import get from 'lodash/get'
+import { generateWebLink, useCapabilities } from 'cozy-client'
+
+const NewRoute = () => {
+  const client = useClient()
+  const { capabilities } = useCapabilities(client)
+  const isFlatDomain = get(capabilities, 'flat_subdomains')
+
+  useEffect(() => {
+    const createNote = async () => {
+      const { data: doc } = await createNoteDocument(client)
+      const returnUrl = generateWebLink({
+        slug: 'notes',
+        cozyUrl: client.getStackClient().uri,
+        subDomainType: isFlatDomain ? 'flat' : 'nested',
```

Comment:
`useCapabilities` is async. So you need to wait the value before calling the createNote() method. If not, isFlatDomain will always be undefined